### PR TITLE
fix(ci): Fix migration checks

### PR DESCRIPTION
**Pull request by armenzg**
This fixes a regression introduced in #25499.

When `sentry django makemigrations --check` does not exit with exit code zero, the
conditional checks in the next line will not be evaluated.

This captures the stdout/stderr into a variable and the exit code into another one.

### CODE REVIEW

**Comment by armenzg** [comment]
We need CI tests 😞

File: scripts/migration-check.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (armenzg) [inline scripts/migration-check.sh:1]: Moving it to a file also helps me verify the check locally.

File: .github/workflows/check-if-migration-is-required.yml [hunk 2]
```diff
@@ -40,15 +41,9 @@ jobs:
...
           PGPASSWORD: postgres
         run: |
-          # Below will exit with non-zero status if model changes are missing migrations
-          sentry django makemigrations --check --dry-run --no-input
-          if [ "$?" == 1 ]; then
-            echo >&2 "::error::Error: Migration required -- to generate a migration, run `sentry django makemigrations -n <some_name> && git add migrations_lockfile.txt`" && exit 1
-          else:
-            echo >&2 "::error::Error: Migration lockfile mismatch -- run `sentry django makemigrations -n <some_name> && git add migrations_lockfile.txt`" && exit 1
-          fi
+          ./scripts/migration-check.sh
```
Comment (billyvg) [inline .github/workflows/check-if-migration-is-required.yml:49]: Should we move this to `.github/workflows/scripts` to indicate that it's only used for CI?

Comment (armenzg) [reply .github/workflows/check-if-migration-is-required.yml:49]: That's a fine idea!

**Review by billyvg — APPROVED** [review]

**Pull request merged**